src/main/java/hudson/plugins/gradle/WithGradleExecution.java

```diff
@@ -40,6 +45,21 @@ public BuildScanCallback(GradleTaskListenerDecorator decorator, StepContext pare
             this.parentContext = parentContext;
         }
 
+        @Override
+        public void onStart(StepContext context) {
```

Comment:
I think this only works if you already use `withGradle` in your Jenkinsfile, right? Though when you have a simple `sh ./gradlew` step which isn't wrapped in `withGradle`, then I think this won't work. Is it enough if it works when you already use `withGradle`?